### ISSUE TITLE
Domain settings

### ISSUE THREAD

**Issue by Slion** [Slion/Fulguris#159]
Implement persistent domain settings including things like:

- Ability to [delete cookies from a domain][1]
- Ability to show cookies from a domain
- Persist WebRTC permissions choices
- Integrate third-party app settings, see #138 
- Notifications settings once we have them working

[1]: https://stackoverflow.com/questions/35390590/remove-all-cookies-from-specific-domain-in-web-view

**Comment by ashwebcode** [comment]
I would like to suggest a domain setting to always keep cookies from a domain. So if you have cookies from this domain, it is excluded from deletion. This would be useful for staying logged into certain trusted sites

**Comment by moya2162** [comment]
This is a great feature to add. [Privacy Browser](https://f-droid.org/en/packages/com.stoutner.privacybrowser.standard/) has this feature and the project is [open source](https://gitweb.stoutner.com/?p=PrivacyBrowserAndroid.git;a=summary), so maybe some of the code can be used as a starting point for this project.

**Comment by Justauser0** [comment]
Will this feature come in near future

**Comment by Cwpute** [comment]
I'd like to add, if Reader mode could be enabled or disabled per-domain that would be awesome.

**Comment by Slion** [comment]
There is some on going work on that feature here: https://github.com/Slion/Fulguris/tree/domain-settings-v2

**Comment by Slion** [comment]
> I'd like to add, if Reader mode could be enabled or disabled per-domain that would be awesome.

Unfortunately that would not work with the current reader mode implementation which is going to through another activity. I suggest you open a specific issue if you want to see a rework of reader mode integrated as tab in the browser activity.

**Comment by Charles7z** [comment]
Good first party cookie management is a big deal not only for privacy but ease of use. 

Being able to enable only per domain is a good feature.

Another idea is auto delete all cookies, if enabled to accept cookies, except for protected domains that are set to accept cookies and possibly have a setting to "protect" that domains cookies — with kiwi browser extensions were easy to set this up, thus sites function fine because cookies are accepted but as soon as tab closed cookies for that site are cleared unless whitelisted. Good privacy + ease of use + site functional. No browsers offer this, only possibly via extensions — clearly people use it.

**Comment by The-First-King** [comment]
Domain search feature would be nice to add into the first post in case this feature might appear one day in the future.

At this moment, both Domains and Resource domains sub-menus do not have search and when a thousand entries are in the list it’s a challenge to find there what is needed.

From my opinion, without having the search, the whole idea about Domain is not usable and the easiest way is just using Default domains settings.

**Comment by Slion** [comment]
@The-First-King 
I agree, search feature would be nice to have, and yes once you have a lot of domains the domains lists are hard to navigate. However there is one easy way to access a specific domain settings without search and without going through the lists. While visiting a domain from the web page go to Menu > Options > Domain.

**Comment by The-First-King** [comment]
This is a partial workaround because, for resource domains, it is not applicable. For example, if we would like to disable/enable JavaScript for some particular resource domains, then it would be a real challenge. Usually behind one domain there are several resource domains. The list of resource domains is approximately 3-4 times larger them main domains.
